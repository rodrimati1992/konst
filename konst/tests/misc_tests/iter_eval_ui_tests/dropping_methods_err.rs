// testing that methods which drop items cause iterators over drop types to error in const


const A: [String; 1] = [String::new()];

const _: () = _ = konst::iter::eval!(A, filter(|_| false),next());
const _: () = _ = konst::iter::eval!(A, map_while(|_| None::<u8>),next());
const _: () = _ = konst::iter::eval!(A, skip_while(|_| false),next());
const _: () = _ = konst::iter::eval!(A, skip(1),next());
const _: () = _ = konst::iter::eval!(A, step_by(1),next());
const _: () = _ = konst::iter::eval!(A, take_while(|_| false),next());
const _: () = _ = konst::iter::eval!(A, take(1),next());
const _: () = _ = konst::iter::eval!(0.., zip(A),next());


fn main() {}