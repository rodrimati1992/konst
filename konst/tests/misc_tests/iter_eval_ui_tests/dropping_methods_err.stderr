error[E0493]: destructor of `String` cannot be evaluated at compile-time
 --> tests/misc_tests/iter_eval_ui_tests/dropping_methods_err.rs:6:19
  |
6 | const _: () = _ = konst::iter::eval!(A, filter(|_| false),next());
  |                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  |                   |
  |                   the destructor for this type cannot be evaluated in constants
  |                   value is dropped here
  |
  = note: this error originates in the macro `$crate::__iter2_interpreter` which comes from the expansion of the macro `konst::iter::eval` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0493]: destructor of `Option<String>` cannot be evaluated at compile-time
 --> tests/misc_tests/iter_eval_ui_tests/dropping_methods_err.rs:6:19
  |
6 | const _: () = _ = konst::iter::eval!(A, filter(|_| false),next());
  |                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  |                   |
  |                   the destructor for this type cannot be evaluated in constants
  |                   value is dropped here
  |
  = note: this error originates in the macro `$crate::__iter2_drop_iterator` which comes from the expansion of the macro `konst::iter::eval` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0493]: destructor of `Option<String>` cannot be evaluated at compile-time
 --> tests/misc_tests/iter_eval_ui_tests/dropping_methods_err.rs:6:19
  |
6 | const _: () = _ = konst::iter::eval!(A, filter(|_| false),next());
  |                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^-
  |                   |                                             |
  |                   |                                             value is dropped here
  |                   the destructor for this type cannot be evaluated in constants
  |
  = note: this error originates in the macro `$crate::__iter2_finish_lowering` which comes from the expansion of the macro `konst::iter::eval` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0493]: destructor of `String` cannot be evaluated at compile-time
 --> tests/misc_tests/iter_eval_ui_tests/dropping_methods_err.rs:7:19
  |
7 | const _: () = _ = konst::iter::eval!(A, map_while(|_| None::<u8>),next());
  |                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  |                   |
  |                   the destructor for this type cannot be evaluated in constants
  |                   value is dropped here
  |
  = note: this error originates in the macro `$crate::__iter2_interpreter` which comes from the expansion of the macro `konst::iter::eval` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0493]: destructor of `Option<String>` cannot be evaluated at compile-time
 --> tests/misc_tests/iter_eval_ui_tests/dropping_methods_err.rs:7:19
  |
7 | const _: () = _ = konst::iter::eval!(A, map_while(|_| None::<u8>),next());
  |                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  |                   |
  |                   the destructor for this type cannot be evaluated in constants
  |                   value is dropped here
  |
  = note: this error originates in the macro `$crate::__iter2_drop_iterator` which comes from the expansion of the macro `konst::iter::eval` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0493]: destructor of `String` cannot be evaluated at compile-time
 --> tests/misc_tests/iter_eval_ui_tests/dropping_methods_err.rs:8:19
  |
8 | const _: () = _ = konst::iter::eval!(A, skip_while(|_| false),next());
  |                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  |                   |
  |                   the destructor for this type cannot be evaluated in constants
  |                   value is dropped here
  |
  = note: this error originates in the macro `$crate::__iter2_interpreter` which comes from the expansion of the macro `konst::iter::eval` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0493]: destructor of `Option<String>` cannot be evaluated at compile-time
 --> tests/misc_tests/iter_eval_ui_tests/dropping_methods_err.rs:8:19
  |
8 | const _: () = _ = konst::iter::eval!(A, skip_while(|_| false),next());
  |                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  |                   |
  |                   the destructor for this type cannot be evaluated in constants
  |                   value is dropped here
  |
  = note: this error originates in the macro `$crate::__iter2_drop_iterator` which comes from the expansion of the macro `konst::iter::eval` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0493]: destructor of `Option<String>` cannot be evaluated at compile-time
 --> tests/misc_tests/iter_eval_ui_tests/dropping_methods_err.rs:8:19
  |
8 | const _: () = _ = konst::iter::eval!(A, skip_while(|_| false),next());
  |                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^-
  |                   |                                                 |
  |                   |                                                 value is dropped here
  |                   the destructor for this type cannot be evaluated in constants
  |
  = note: this error originates in the macro `$crate::__iter2_finish_lowering` which comes from the expansion of the macro `konst::iter::eval` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0493]: destructor of `String` cannot be evaluated at compile-time
 --> tests/misc_tests/iter_eval_ui_tests/dropping_methods_err.rs:9:19
  |
9 | const _: () = _ = konst::iter::eval!(A, skip(1),next());
  |                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  |                   |
  |                   the destructor for this type cannot be evaluated in constants
  |                   value is dropped here
  |
  = note: this error originates in the macro `$crate::__iter2_interpreter` which comes from the expansion of the macro `konst::iter::eval` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0493]: destructor of `Option<String>` cannot be evaluated at compile-time
 --> tests/misc_tests/iter_eval_ui_tests/dropping_methods_err.rs:9:19
  |
9 | const _: () = _ = konst::iter::eval!(A, skip(1),next());
  |                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  |                   |
  |                   the destructor for this type cannot be evaluated in constants
  |                   value is dropped here
  |
  = note: this error originates in the macro `$crate::__iter2_drop_iterator` which comes from the expansion of the macro `konst::iter::eval` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0493]: destructor of `Option<String>` cannot be evaluated at compile-time
 --> tests/misc_tests/iter_eval_ui_tests/dropping_methods_err.rs:9:19
  |
9 | const _: () = _ = konst::iter::eval!(A, skip(1),next());
  |                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^-
  |                   |                                   |
  |                   |                                   value is dropped here
  |                   the destructor for this type cannot be evaluated in constants
  |
  = note: this error originates in the macro `$crate::__iter2_finish_lowering` which comes from the expansion of the macro `konst::iter::eval` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0493]: destructor of `String` cannot be evaluated at compile-time
  --> tests/misc_tests/iter_eval_ui_tests/dropping_methods_err.rs:10:19
   |
10 | const _: () = _ = konst::iter::eval!(A, step_by(1),next());
   |                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |                   |
   |                   the destructor for this type cannot be evaluated in constants
   |                   value is dropped here
   |
   = note: this error originates in the macro `$crate::__iter2_interpreter` which comes from the expansion of the macro `konst::iter::eval` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0493]: destructor of `Option<String>` cannot be evaluated at compile-time
  --> tests/misc_tests/iter_eval_ui_tests/dropping_methods_err.rs:10:19
   |
10 | const _: () = _ = konst::iter::eval!(A, step_by(1),next());
   |                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |                   |
   |                   the destructor for this type cannot be evaluated in constants
   |                   value is dropped here
   |
   = note: this error originates in the macro `$crate::__iter2_drop_iterator` which comes from the expansion of the macro `konst::iter::eval` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0493]: destructor of `Option<String>` cannot be evaluated at compile-time
  --> tests/misc_tests/iter_eval_ui_tests/dropping_methods_err.rs:10:19
   |
10 | const _: () = _ = konst::iter::eval!(A, step_by(1),next());
   |                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^-
   |                   |                                      |
   |                   |                                      value is dropped here
   |                   the destructor for this type cannot be evaluated in constants
   |
   = note: this error originates in the macro `$crate::__iter2_finish_lowering` which comes from the expansion of the macro `konst::iter::eval` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0493]: destructor of `String` cannot be evaluated at compile-time
  --> tests/misc_tests/iter_eval_ui_tests/dropping_methods_err.rs:11:19
   |
11 | const _: () = _ = konst::iter::eval!(A, take_while(|_| false),next());
   |                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |                   |
   |                   the destructor for this type cannot be evaluated in constants
   |                   value is dropped here
   |
   = note: this error originates in the macro `$crate::__iter2_interpreter` which comes from the expansion of the macro `konst::iter::eval` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0493]: destructor of `Option<String>` cannot be evaluated at compile-time
  --> tests/misc_tests/iter_eval_ui_tests/dropping_methods_err.rs:11:19
   |
11 | const _: () = _ = konst::iter::eval!(A, take_while(|_| false),next());
   |                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |                   |
   |                   the destructor for this type cannot be evaluated in constants
   |                   value is dropped here
   |
   = note: this error originates in the macro `$crate::__iter2_drop_iterator` which comes from the expansion of the macro `konst::iter::eval` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0493]: destructor of `Option<String>` cannot be evaluated at compile-time
  --> tests/misc_tests/iter_eval_ui_tests/dropping_methods_err.rs:11:19
   |
11 | const _: () = _ = konst::iter::eval!(A, take_while(|_| false),next());
   |                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^-
   |                   |                                                 |
   |                   |                                                 value is dropped here
   |                   the destructor for this type cannot be evaluated in constants
   |
   = note: this error originates in the macro `$crate::__iter2_finish_lowering` which comes from the expansion of the macro `konst::iter::eval` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0493]: destructor of `String` cannot be evaluated at compile-time
  --> tests/misc_tests/iter_eval_ui_tests/dropping_methods_err.rs:12:19
   |
12 | const _: () = _ = konst::iter::eval!(A, take(1),next());
   |                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |                   |
   |                   the destructor for this type cannot be evaluated in constants
   |                   value is dropped here
   |
   = note: this error originates in the macro `$crate::__iter2_interpreter` which comes from the expansion of the macro `konst::iter::eval` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0493]: destructor of `Option<String>` cannot be evaluated at compile-time
  --> tests/misc_tests/iter_eval_ui_tests/dropping_methods_err.rs:12:19
   |
12 | const _: () = _ = konst::iter::eval!(A, take(1),next());
   |                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |                   |
   |                   the destructor for this type cannot be evaluated in constants
   |                   value is dropped here
   |
   = note: this error originates in the macro `$crate::__iter2_drop_iterator` which comes from the expansion of the macro `konst::iter::eval` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0493]: destructor of `Option<String>` cannot be evaluated at compile-time
  --> tests/misc_tests/iter_eval_ui_tests/dropping_methods_err.rs:12:19
   |
12 | const _: () = _ = konst::iter::eval!(A, take(1),next());
   |                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^-
   |                   |                                   |
   |                   |                                   value is dropped here
   |                   the destructor for this type cannot be evaluated in constants
   |
   = note: this error originates in the macro `$crate::__iter2_finish_lowering` which comes from the expansion of the macro `konst::iter::eval` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0493]: destructor of `Option<String>` cannot be evaluated at compile-time
  --> tests/misc_tests/iter_eval_ui_tests/dropping_methods_err.rs:13:19
   |
13 | const _: () = _ = konst::iter::eval!(0.., zip(A),next());
   |                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |                   |
   |                   the destructor for this type cannot be evaluated in constants
   |                   value is dropped here
   |
   = note: this error originates in the macro `$crate::__iter2_drop_iterator` which comes from the expansion of the macro `konst::iter::eval` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0493]: destructor of `Option<(i32, String)>` cannot be evaluated at compile-time
  --> tests/misc_tests/iter_eval_ui_tests/dropping_methods_err.rs:13:19
   |
13 | const _: () = _ = konst::iter::eval!(0.., zip(A),next());
   |                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^-
   |                   |                                    |
   |                   |                                    value is dropped here
   |                   the destructor for this type cannot be evaluated in constants
   |
   = note: this error originates in the macro `$crate::__iter2_finish_lowering` which comes from the expansion of the macro `konst::iter::eval` (in Nightly builds, run with -Z macro-backtrace for more info)
